### ISSUE TITLE
XRT reader adds non-standard value for FITS keyword

### ISSUE THREAD

**Issue by wtbarnes** [sunpy/sunpy#2405]
The [current XRT map reader](https://github.com/sunpy/sunpy/blob/master/sunpy/map/sources/hinode.py#L24) adds the `wavelnth` keyword to the map metadata and populates it with the value `np.nan`. This is a problem because `nan` is a non-standard value for a FITS keyword so when trying to save a map, Astropy will raise an error.

To reproduce this error, download your favorite XRT FITS file `my_xrt_obs.fits` and then,

```python
m = Map('my_xrt_obs.fits')
print(m.meta['wavelnth']) # should show nan
m.save() # should raise an error about non-standard FITS keywords
```

which raises the error,
```python
ValueError: Floating point nan values are not allowed in FITS headers.
```

The question then is what should the value of this keyword be? Does `Map` need a `wavelnth` keyword at all? The keyword doesn't make much sense in the context of XRT observations as the "channels" are really denoted by different filter wheel combinations (which are already recorded in the metadata) so the `wavelnth` keyword is kind of meaningless anyway.

If `sunpy.map.GenericMap` doesn't require it, I would suggest removing the `wavelnth` keyword altogether.

**Comment by getsanjeev** [comment]
I did grep in /../../sunpy/ to find usage of "wavelnth" in the project. This is what I got : 

sunpy/map/tests/test_mapbase.py:        'wavelnth': 10,
sunpy/map/tests/test_mapbase.py:            'wavelnth': 10,
sunpy/map/tests/test_mapbase.py:        'wavelnth': 10,

sunpy/map/mapbase.py:        return u.Quantity(self.meta.get('wavelnth', 0),
sunpy/map/mapbase.py:        return u.Quantity(self.meta.get('wavelnth', 0),
sunpy/map/sources/soho.py:        self.meta['wavelnth'] = np.nan
sunpy/map/sources/soho.py:        self.meta['wavelnth'] = np.nan
sunpy/map/sources/iris.py:        self.meta['wavelnth'] = header['twave1'] ===> Not so sure about this 
sunpy/map/sources/hinode.py:        self.meta['wavelnth'] = np.nan
sunpy/map/sources/rhessi.py:        self.meta['wavelnth'] = [self.meta['energy_l'], self.meta['energy_h']] ===> Not so sure about this
sunpy/map/sources/stereo.py:        self.meta['wavelnth'] = np.nan
sunpy/map/sources/stereo.py:        self.meta['wavelnth'] = np.nan
sunpy/map/sources/yohkoh.py:        s = self.meta.get('wavelnth', '')

Binary file sunpy/map/sources/__pycache__/hinode.cpython-36.pyc matches
Binary file sunpy/map/sources/__pycache__/rhessi.cpython-36.pyc matches
Binary file sunpy/map/sources/__pycache__/yohkoh.cpython-36.pyc matches
Binary file sunpy/map/sources/__pycache__/soho.cpython-36.pyc matches
Binary file sunpy/map/sources/__pycache__/iris.cpython-36.pyc matches
Binary file sunpy/map/sources/__pycache__/stereo.cpython-36.pyc matches
Binary file sunpy/map/__pycache__/mapbase.cpython-36.pyc matches

sunpy/io/tests/test_fits.py:def test_extract_waveunit_missing_waveunit_key_and_missing_wavelnth_comment():
sunpy/io/tests/test_fits.py:def test_missing_waveunit_in_wavelnth_comment():
sunpy/io/tests/test_fits.py:def test_extract_waveunit_wavelnthcomment_brackets():
sunpy/io/tests/test_fits.py:def test_extract_waveunit_wavelnthcomment_parentheses():

sunpy/io/fits.py:    # 3. parse wavelnth_comment
sunpy/io/fits.py:    wavelnth_comment = header['KEYCOMMENTS'].get('WAVELNTH')
sunpy/io/fits.py:    elif wavelnth_comment is not None:
sunpy/io/fits.py:            m = re.search(pattern, wavelnth_comment)

Binary file sunpy/io/__pycache__/fits.cpython-36.pyc matches
sunpy/coordinates/tests/test_wcs_utils.py:              'wavelnth': 10

**Comment by getsanjeev** [comment]
I did grep in /../../sunpy/ to find usage of "wavelnth" in the project. This is what I got :

sunpy/map/tests/test_mapbase.py: 'wavelnth': 10,
sunpy/map/tests/test_mapbase.py: 'wavelnth': 10,
sunpy/map/tests/test_mapbase.py: 'wavelnth': 10,

sunpy/map/mapbase.py: return u.Quantity(self.meta.get('wavelnth', 0),
sunpy/map/mapbase.py: return u.Quantity(self.meta.get('wavelnth', 0),
sunpy/map/sources/soho.py: self.meta['wavelnth'] = np.nan
sunpy/map/sources/soho.py: self.meta['wavelnth'] = np.nan
sunpy/map/sources/iris.py: self.meta['wavelnth'] = header['twave1'] ===> Not so sure about this
sunpy/map/sources/hinode.py: self.meta['wavelnth'] = np.nan
sunpy/map/sources/rhessi.py: self.meta['wavelnth'] = [self.meta['energy_l'], self.meta['energy_h']] ===> Not so sure about this
sunpy/map/sources/stereo.py: self.meta['wavelnth'] = np.nan
sunpy/map/sources/stereo.py: self.meta['wavelnth'] = np.nan
sunpy/map/sources/yohkoh.py: s = self.meta.get('wavelnth', '')

Binary file sunpy/map/sources/pycache/hinode.cpython-36.pyc matches
Binary file sunpy/map/sources/pycache/rhessi.cpython-36.pyc matches
Binary file sunpy/map/sources/pycache/yohkoh.cpython-36.pyc matches
Binary file sunpy/map/sources/pycache/soho.cpython-36.pyc matches
Binary file sunpy/map/sources/pycache/iris.cpython-36.pyc matches
Binary file sunpy/map/sources/pycache/stereo.cpython-36.pyc matches
Binary file sunpy/map/pycache/mapbase.cpython-36.pyc matches

sunpy/io/tests/test_fits.py:def test_extract_waveunit_missing_waveunit_key_and_missing_wavelnth_comment():
sunpy/io/tests/test_fits.py:def test_missing_waveunit_in_wavelnth_comment():
sunpy/io/tests/test_fits.py:def test_extract_waveunit_wavelnthcomment_brackets():
sunpy/io/tests/test_fits.py:def test_extract_waveunit_wavelnthcomment_parentheses():

sunpy/io/fits.py: # 3. parse wavelnth_comment
sunpy/io/fits.py: wavelnth_comment = header['KEYCOMMENTS'].get('WAVELNTH')
sunpy/io/fits.py: elif wavelnth_comment is not None:
sunpy/io/fits.py: m = re.search(pattern, wavelnth_comment)

Binary file sunpy/io/pycache/fits.cpython-36.pyc matches
sunpy/coordinates/tests/test_wcs_utils.py: 'wavelnth': 10

**Comment by getsanjeev** [comment]
@wtbarnes @nabobalis  It seems clearly that except for few (iris and rhessi) we are not currently utilising "wavelnth" keyword (populating them nan) and that will cause an error as mentioned in the issue. We have two ways to go about it --
1. Remove this meta data keyword from the ones we are [not using & setting them nan].
2. Set some default value, may be negative number and check them while reading. (This would maintain the consistency). But then in map yohkoh.py we have not populated it.

How should we go about it?

**Comment by Cadair** [comment]
It was agreed at the meeting that we should just remove any instance of `meta['wavelnth'] = nan`. i.e. not have 'wavelnth' in the header. This is because the header should be FITS compliant and `GenericMap` will handle missing 'wavelnth' gracefully.

**Comment by getsanjeev** [comment]
@Cadair  So I just need to remove these instances of self.meta['wavelnth'] = np.nan from all the maps (derived from GenericMap). RIght?

**Comment by Cadair** [comment]
Yes please

**Comment by wtbarnes** [comment]
Fixed by #2427 and #2456
